### ISSUE TITLE
Add support for PermissionsBoundaries for Roles

### ISSUE THREAD

**Issue by JordanDeBeer** [aws-quickstart/quickstart-amazon-eks-cluster-resource-provider#3]
Some organizations require [Permissions Boundaries](https://docs.aws.amazon.com/IAM/latest/UserGuide/access_policies_boundaries.html) to be present on all IAM Roles created.  It would be useful if this was an option that could be passed into the stack at installation time.

This could potentially be accomplished via a Conditional.

**Comment by JordanDeBeer** [comment]
Closed by #4

**Comment by mattiassluis** [comment]
Hi Jordan, would it be possible to provide a description on how to use this? I am using the quickstart-amazon-eks template but I have no idea on how to provide the permissions boundary (I guess I have to provide this as part of the RegisterType but I can't figure out how)

**Comment by JordanDeBeer** [comment]
Hey @mattiassluis, when you create the stack per the README, you can pass in a Parameter for your PermissionsBoundary. The key is `ParameterKey` and the value should be the ARN of your Permissions Boundary. 

An example _could_ look like this.

```
aws cloudformation create-stack \
  --stack-name awsqs-eks-cluster-resource \
  --capabilities CAPABILITY_NAMED_IAM \
  --template-url https://s3.amazonaws.com/aws-quickstart/quickstart-amazon-eks-cluster-resource-provider/deploy.template.yaml \
  --region us-west-2 \
  --parameters ParameterKey=CreateClusterAccessRole,ParameterValue='true' ParameterKey=PermissionsBoundary,ParameterValue=arn:aws::iam::YOURAWSACCOUNTID:policy/YOURPERMISSIONSBOUNDARY
```

**Comment by mattiassluis** [comment]
Hi @JordanDeBeer thanks for your response! I noticed that later but realized that this resource is not deployed in that way in the quickstart for aws (which also lacks support for the PermissionsBoundary). Most of the roles created by this stack are created in other ways but I solved it in my case by added the KubernetesVpcRole definition to the amazon-eks-iam.template.yaml (a file I am already modifying heavily to accommodate my need for the PermissionsBoundary support). With the resource already existing the go module doesn't try to create it.
